src/pytorch_kinematics/transforms/rotation_conversions.py

```diff
@@ -449,12 +450,64 @@ def quaternion_apply(quaternion, point):
     return out[..., 1:]
 
 
-def axis_and_angle_to_matrix_directly(axis, theta):
+def tensor_axis_and_d_to_pris_matrix(axis, d):
+    """
+    Works with any number of batch dimensions.
+
+    Args:
+        axis: [..., 3]
+        d: [ ...]
+
+    Returns: [..., 4, 4]
+
+    """
+    batch_axes = axis.shape[:-1]
+    mat44 = torch.eye(4).to(axis).repeat(*batch_axes, 1, 1)
+    pos = axis * d[..., None]
+    mat44[..., :3, 3] = pos
```

Comment:
```suggestion
    mat33 = torch.eye(3).to(axis).expand(*batch_axes, 3, 3)
    pos = axis * d.unsqueeze(-1)
    mat44 = torch.cat((mat33, pos.unsqueeze(-1)), -1)
    mat44 = torch.cat((mat44, torch.tensor([0.0, 0.0, 0.0, 1.0]).expand(*batch_axes, 1, 4).to(axis)), -2)
```

Replace in-place operations for vmap